modules/public/views.py

```diff
@@ -151,6 +152,10 @@ def snap_details(snap_name):
     with appropriate sanitation.
     """
 
+    pattern = re.compile("^[a-z0-9-]*[a-z][a-z0-9-]*$")
+    if not pattern.match(snap_name):
```

Comment:
It would be neater to use the Regex URL converter, as [used in insights.ubuntu.com](https://github.com/canonical-websites/insights.ubuntu.com/blob/master/app.py#L22), so that it will only hit this route for valid snap names - then other URLs will 404 automatically.